packages/cli/__tests__/daemon/__integration__/cli.zowe.daemon.integration.suite.test.ts

```diff
@@ -0,0 +1,97 @@
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+/* Our daemon-related tests create a TGZ for our EXE, which should be removed
+ * after testing. These tests also create and remove an EXE file in our bin
+ * directory. The file creations and deletions create problems when the tests
+ * are run in parallel. By placing the daemon tests within this top-level
+ * describe (thus contining the tests in a single file) the tests are serialized.
+ */
+
+import * as nodeJsPath from "path";
+import * as tar from "tar";
+import { IO, ProcessUtils, ISystemInfo } from "@zowe/imperative";
+
+describe("Zowe daemon suite", () => {
+    const rimrafSync = require("rimraf").sync;
+
+    let prebuildsDir: string;
+    let zoweExePath: string;
+    let zoweExeDirPath: string;
+    let zoweExeTgzPath: string;
+    let zoweExeFileNm: string;
+
+    // establish path names and record whether we are running our EXE or node.js script
+    beforeAll(async () => {
+        // determine executable file name and TGZ path for our current OS
+        const zoweRootDir: string = nodeJsPath.normalize(__dirname + "/../../../../..");
+        prebuildsDir = nodeJsPath.normalize(zoweRootDir + "/packages/cli/prebuilds");
+        const sysInfo: ISystemInfo = ProcessUtils.getBasicSystemInfo();
+        switch (sysInfo.platform) {
+            case "darwin": {
+                zoweExeFileNm = "zowe";
+                zoweExeTgzPath = nodeJsPath.resolve(prebuildsDir, "zowe-macos.tgz");
+                break;
+            }
+            case "linux": {
+                zoweExeFileNm = "zowe";
+                zoweExeTgzPath = nodeJsPath.resolve(prebuildsDir, "zowe-linux.tgz");
+                break;
+            }
+            case "win32": {
+                zoweExeFileNm = "zowe.exe";
+                zoweExeTgzPath = nodeJsPath.resolve(prebuildsDir, "zowe-windows.tgz");
+                break;
+            }
+            default: {
+                zoweExeFileNm = "exeForUnknownOs";
+                zoweExeTgzPath += "unknownOs.tgz";
+                throw "cli.zowe.daemon.integration.suite.test.ts: beforeAll: " +
+                    sysInfo.platform + " is not a known OS.";
+            }
+        }
+
+        // Form the path name to our executable.
+        zoweExeDirPath = nodeJsPath.normalize(zoweRootDir + "/zowex/target/release");
+        zoweExePath = nodeJsPath.resolve(zoweExeDirPath, zoweExeFileNm);
+        if (!IO.existsSync(zoweExePath)) {
+            zoweExeDirPath = nodeJsPath.normalize(zoweRootDir + "/zowex/target/debug");
+            zoweExePath = nodeJsPath.resolve(zoweExeDirPath, zoweExeFileNm);
+            if (!IO.existsSync(zoweExePath)) {
+                zoweExePath = "./NoZoweExeExists";
+            }
```

Comment:
Is this code used? If not, should the `let zoweExePath` on line 27 be removed and replaced with `const zoweExePath` on line 64?

Reply:
I believe the code will be used. zoweExeDirPath is assigned the value "...release" and later the value "...debug". So, the two assignments to zoweExePath are both different. The intent is to use the "release" version of the EXE if it exists, and then fallback to the "debug" version of the EXE, to ensure that the test will also run on a developer's laptop.

However, I will make the declaration of the variables more local to their use. I also noticed that the assignment of "./NoZoweExeExists" was to a variable not used by the later tar command, so I changed that assignment. If no EXE is built during our build, I want the tar command to fail in a very obvious way.

Reply:
The changes in 6a267aecaff36906501a9b44d75bbc5a5313e71a addressed my concerns about unused variables.